python/src/iceberg/conversions.py

```diff
@@ -0,0 +1,294 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Utility module for various conversions around PrimitiveType implementations
+
+This module enables:
+    - Converting partition strings to built-in python objects
+    - Converting a value to a byte buffer
+    - Converting a byte buffer to a value
+
+Note:
+    Conversion logic varies based on the PrimitiveType implementation. Therefore conversion functions
+    are defined here as generic functions using the @singledispatch decorator. For each PrimitiveType
+    implementation, a concrete function is registered for each generic conversion function. For PrimitiveType
+    implementations that share the same conversion logic, registrations can be stacked.
+"""
+import struct
+import uuid
+from decimal import Decimal, localcontext
+from functools import singledispatch
+from typing import Union
+
+from iceberg.types import (
+    BinaryType,
+    BooleanType,
+    DateType,
+    DecimalType,
+    DoubleType,
+    FixedType,
+    FloatType,
+    IntegerType,
+    LongType,
+    PrimitiveType,
+    StringType,
+    TimestampType,
+    TimestamptzType,
+    TimeType,
+    UUIDType,
+)
+
+
+def convert_decimal_to_unscaled_value(decimal_type_var: DecimalType, value: Decimal) -> int:
```

Comment:
What about a bit simpler name? This could be `decimal_to_unscaled`. The "convert" is implied by being in `conversions` and "value" is unnecessary because this clearly returns a value.

Reply:
Haha yeah that was a pretty ugly function name 😅 . Updated!